### PR TITLE
Create run-tests-draft.md - no-merge (WIP)

### PR DESCRIPTION
**Pull request by NikolaMilosavljevic**

### CODE REVIEW

**Comment by omajid** [comment]
Does this mean that there will be 2 ways to build source build:

- Build sources only. This step will **not** download prebuilts
- Build sources and/or run tests. This step may download prebuilts and run tests?

**Comment by NikolaMilosavljevic** [comment]
> Does this mean that there will be 2 ways to build source build:
> 
> * Build sources only. This step will **not** download prebuilts
> * Build sources and/or run tests. This step may download prebuilts. All tests from all projects will be run (including end-to-end cli tests).

Building and running test will work with toolsets built from sources, no prebuilts.

File: Documentation/run-tests-draft.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+### Modify `build.cmd/sh/ps1`
+
+Skip submodule check for `BuildTests` and `RunTests` options
+
+### Modify `build.proj`
+
+Add new targets for building and running tests, i.e. `BuildTests` and `RunTests`.
+
+These targets should run msbuild of RootRepoTests with appropriate parameters, i.e. `BuildTests`, `RunTests`, skip repo references (`SkipRepoReferences=true`)
```
Comment (dseefeld) [inline Documentation/run-tests-draft.md]: Probably SkipPatches=true as well.

File: Documentation/run-tests-draft.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+```xml
+<BuildArguments>$(BuildArguments) /p:SkipTests=true</BuildArguments>
+```
+
+Add:
+
+```xml
+<BuildArguments Condition="'$(RunTests)' != 'true'">$(BuildArguments) /p:SkipTests=true</BuildArguments>
+<BuildArguments Condition="'$(RunTests)' != 'true'">$(BuildArguments) /p:BuildTests=true</BuildArguments>
```
Comment (crummel) [inline Documentation/run-tests-draft.md]: Should this be `RunTests == true` => `BuildTests=true` or `RunTests != true` => `BuildTests=false`?  It looks like what we're doing currently is building tests every time we're not running them, which would happen during the normal build, right?

File: Documentation/run-tests-draft.md (hunk 1 above)
Comment (NikolaMilosavljevic) [reply Documentation/run-tests-draft.md]: Yes, I'll add that.

File: Documentation/run-tests-draft.md (hunk 2 above)
Comment (NikolaMilosavljevic) [reply Documentation/run-tests-draft.md]: At the moment I can only enable test-build as part of CoreFX build target, which would build everything, including tests. Once we get newer CoreFx changes, we would modify it to build tests as separate step - we could even combine that step with test-run. Those changes are outlined in the next section starting at line 60.

**Comment by crummel** [comment]
@NikolaMilosavljevic want to merge this or should we close it?

**Comment by NikolaMilosavljevic** [comment]
> @NikolaMilosavljevic want to merge this or should we close it?

This can be closed - it was meant to be used for design discussion. It's also outdated, I believe.

Test documentation can be added to main document. @adaggarwal, would you like to add instructions for running tests to main source-build documentation?

**Comment by crummel** [comment]
I think Aditya already merged that one, it was https://github.com/dotnet/source-build/pull/1263.  Thanks!

**Comment by adaggarwal** [comment]
> Test documentation can be added to main document. @adaggarwal Aditya Aggarwal FTE, would you like to add instructions for running tests to main source-build documentation?

@NikolaMilosavljevic Yep it was added. Feel free to suggest changes if you'd like see some.

**Pull request closed without merging**